src/coreclr/System.Private.CoreLib/src/System/GC.CoreCLR.cs

```diff
@@ -318,7 +319,16 @@ private static unsafe uint RunFinalizers()
                 void* fptr = GetNextFinalizeableObject(ObjectHandleOnStack.Create(ref target));
                 if (fptr == null)
                     break;
-                ((delegate*<object, void>)fptr)(target!);
+
+                try
+                {
+                    ((delegate*<object, void>)fptr)(target!);
+                }
+                catch (Exception ex) when (ExceptionHandling.s_handler != null && ExceptionHandling.s_handler(ex))
```

Comment:
```suggestion
                catch (Exception ex) when (ExceptionHandling.s_handler?.Invoke(ex) ?? false)
```
In case you would like to write this in modern terse C#. (I am not commenting on all places.)

Reply:
started like this, but it resulted in C# complaining about missing nullable types in a few cases, so I reverted to "old-school C#". 
I guess I could use fluent syntax in places where it works.